benchmark/lib/Streamly/Benchmark/Common.hs

```diff
@@ -134,76 +134,20 @@ defaultStreamSize = 100000
 
-newtype BenchOpts = StreamSize Int deriving Show
-
 getStreamSize :: String -> Int
 getStreamSize size =
     case (readMaybe size :: Maybe Int) of
         Just x -> x
         Nothing -> error "Stream size must be numeric"
 
-options :: [OptDescr BenchOpts]
-options =
-    [
-      Option [] ["stream-size"] (ReqArg getSize "COUNT") "Stream element count"
-    ]
-
-    where
-
-    getSize = StreamSize . getStreamSize
-
-deleteOptArgs
-    :: (Maybe String, Maybe String) -- (prev, yielded)
-    -> String
-    -> (Maybe String, Maybe String)
-deleteOptArgs (Nothing, _) opt =
-    if opt == "--stream-size"
-    then (Just opt, Nothing)
-    else (Just opt, Just opt)
-
-deleteOptArgs (Just prev, _) opt =
-    if opt == "--stream-size" || prev == "--stream-size"
-    then (Just opt, Nothing)
-    else (Just opt, Just opt)
-
-parseCLIOpts :: Int -> IO (Int, Config, [String])
-parseCLIOpts defStreamSize = do
-    args <- getArgs
-
-    -- Parse custom options
-    let (opts, _, _, errs) = getOpt' Permute options args
-    when (not $ null errs) $ error $ concat errs
-    (streamSize, args') <-
-        case opts of
-            StreamSize x : _ -> do
-                putStrLn $ "Stream size: " ++ show x
-                -- When using the gauge "--measure-with" option we need to make
-                -- sure that we pass the stream size to child process forked by
-                -- gauge. So we use this env var for that purpose.
-                setEnv "STREAM_SIZE" (show x)
-                -- Hack! remove the option and its argument from args
-                -- getOpt should have a way to return the unconsumed args in
-                -- correct order.
-                newArgs <-
-                          evaluate
-                        $ mapMaybe snd
-                        $ scanl' deleteOptArgs (Nothing, Nothing) args
-                return (x, newArgs)
-            _ -> do
-                r <- lookupEnv "STREAM_SIZE"
-                case r of
-                    Just x -> do
-                        s <- evaluate $ getStreamSize x
-                        return (s, args)
-                    Nothing -> do
-                        setEnv "STREAM_SIZE" (show defStreamSize)
-                        return (defStreamSize, args)
-
-    -- Parse gauge options
-    let config = defaultConfig
-                { timeLimit = Just 1
-                , minDuration = 0
-                , includeFirstIter = streamSize > defStreamSize
-                }
-    let (cfg, benches) = parseWith config args'
-    streamSize `seq` return (streamSize, cfg, benches)
+parseEnvOpts :: Int -> IO Int
+parseEnvOpts defStreamSize = do
+    streamSize <- do
+        r <- lookupEnv "STREAM_SIZE"
+        case r of
+            Just x -> evaluate $ getStreamSize x
+            Nothing -> do
+                setEnv "STREAM_SIZE" (show defStreamSize)
+                return defStreamSize
+
+    streamSize `seq` return streamSize
```

Comment:
Since tasty-bench supports custom CLI args let's revert this change and keep the original `parseCLIOpts` function for gauge, and implement another version of it for tasty-bench. We basically need to pass the stream size via the `--stream-size` CLI arg.

We will also have to change `bench.sh` so that we pass the stream size via CLI instead of passing it via env var.

Reply:
Below code snippet in Readme for Custom command-line options is giving compilation error:
**customOpts & benchIngredients** in the below code snippet are not the same Types.

> main :: IO ()
> main = do
>   let customOpts  = [Option (Proxy :: Proxy RandomSeed)]
>       ingredients = includingOptions **customOpts : benchIngredients**
>   opts <- parseOptions ingredients benchmarks

Reply:
Snippet works for me fine; `customOpts` and `benchIngredients` are different types, but `includingOptions customOpts` and `benchIngredients` are the same.

You can check a complete example at https://github.com/haskell/vector/blob/a735abb688c870cf724bd620a0952e307252c18f/vector/benchmarks/Main.hs#L49